docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -122,13 +127,18 @@ public String toString() {
               .add("commands", commands)
               .add("volumes", volumes)
               .add("env", env)
+              .add("portBindings", portBindings)
               .toString();
    }
 
    public static final DockerTemplateOptions NONE = new DockerTemplateOptions();
 
    public DockerTemplateOptions volumes(Map<String, String> volumes) {
-      this.volumes = Optional.<Map<String, String>> of(ImmutableMap.copyOf(volumes));
+      for (Map.Entry<String, String> entry : checkNotNull(volumes, "volumes").entrySet()) {
+         checkNotNull(entry.getKey(), "host paths must not be null");
+         checkNotNull(entry.getValue(), "container paths must not be null");
+      }
```

Comment:
In general for the whole PR, are these checks really needed? Guava's immutable collections don't allow null values so checking the null input in the map and list options here looks redundant?